backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -294,21 +293,111 @@ public void setUserIsActive(String username, Boolean active) {
     }
   }
 
-  public void createOrganization(Organization org) {
+  public void createOrganization(
+      Organization org, Collection<Facility> facilities, boolean migration) {
     String name = org.getOrganizationName();
     String externalId = org.getExternalId();
+
+    // After the first migration, the migration code below will amount to a no-op,
+    // until we change the next line to new OrganizationRole's
+    Set<OrganizationRole> migrationDestRoles =
+        Set.of(OrganizationRole.NO_ACCESS, OrganizationRole.ALL_FACILITIES);
+    List<String> migrationDestGroupIds = new ArrayList<>();
+    List<String> migrationDestGroupNames = new ArrayList<>();
+    List<String> migrationSourceGroupNames = new ArrayList<>();
+
     for (OrganizationRole role : OrganizationRole.values()) {
-      Group g =
-          GroupBuilder.instance()
-              .setName(generateRoleGroupName(externalId, role))
-              .setDescription(generateRoleGroupDescription(name, role))
-              .buildAndCreate(_client);
-      _app.createApplicationGroupAssignment(g.getId());
+      String roleGroupName = generateRoleGroupName(externalId, role);
+      String roleGroupDescription = generateRoleGroupDescription(name, role);
+      try {
+        Group g =
+            GroupBuilder.instance()
+                .setName(roleGroupName)
+                .setDescription(roleGroupDescription)
+                .buildAndCreate(_client);
+        _app.createApplicationGroupAssignment(g.getId());
+
+        if (migration) {
+          LOG.info("Created Okta group={} via migration", roleGroupName);
+        }
+
+        if (migration && migrationDestRoles.contains(role)) {
+          migrationDestGroupIds.add(g.getId());
+          migrationDestGroupNames.add(roleGroupName);
+        }
+      } catch (ResourceException e) {
+        if (migration) {
+          migrationSourceGroupNames.add(roleGroupName);
+        } else {
+          // ignore attempts to create groups that already exist if this is in migration mode
```

Comment:
Should this comment be moved to the `if` clause? I'm guessing that the call to `migrationSourceGroupNames.add(roleGroupName);` is what constitutes ignoring the exception.